cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/LineageOperationsProcessor.java

```diff
@@ -296,9 +314,19 @@ private List<InputField> createInputFields(List<String> fields, String currentSt
         // first
         String stage = parentsIterator.previous();
         // check if the current field is output by any one of operation created by current stage
-        String origin = stageOutputsWithOrigins.get(stage).get(actualField);
-        if (origin != null) {
-          inputFields.put(field, InputField.of(origin, actualField));
+        List<Origin> possibleOrigins = stageOutputsWithOrigins.get(stage).get(actualField);
+        if (possibleOrigins != null) {
+          // if there is only one origin, update the input field. In this case, tag will be null
```

Comment:
One operation can have multiple origins only if tag is null, otherwise it is original behavior